src/utils/__tests__/use-previous.test.ts

```diff
@@ -0,0 +1,47 @@
+import {renderHook, act} from '@testing-library/react-hooks';
+import {usePrevious} from '../use-previous';
+
+describe('usePrevious', () => {
+  it('should return undefined on initial render', () => {
+    const {result} = renderHook(() => usePrevious(0));
+    expect(result.current).toBeUndefined();
+  });
+
+  it('should return the previous value after update', () => {
+    let value = 0;
+    const {result, rerender} = renderHook(() => usePrevious(value));
+
+    act(() => {
+      value = 1;
+      rerender();
+    });
+
+    expect(result.current).toBe(0);
+
+    act(() => {
+      value = 2;
+      rerender();
+    });
+
+    expect(result.current).toBe(1);
```

Comment:
I think it is preferred (and looks cleaner) to pass the new props into the `rerender` function. Then you are not updating mutable state, and don't need `act`.

```ts
    const {result, rerender} = renderHook((count) => usePrevious(count), {
      initialProps: 0,
    });

    rerender(1);
    expect(result.current).toBe(0);

    rerender(2);
    expect(result.current).toBe(1);
```